templates/outer-loop/gradle/Dockerfile

```diff
@@ -0,0 +1,88 @@
+# Step 1: Build the user's application.
+FROM {{.STACK_IMAGE_GRADLE}} as compile
+
+# Make a well known place for shared library jars separate from the rest of the <server> contents (to help with caching).
+RUN mkdir /work/configlibdir \
+   && mkdir /work/config \
+   &&  mkdir /work/shared
+
+# Copy application source and build data.
+COPY --chown=1001:0 ./. /work/outer-loop-app/tmp/
+RUN cp -r /work/outer-loop-app/tmp/src /work/outer-loop-app/.; \
+    cp -r /work/outer-loop-app/tmp/*.gradle /work/outer-loop-app/tmp/gradle.* /work/outer-loop-app/tmp/gradle* /work/outer-loop-app/. 2>/dev/null | true; \
```

Comment:
The `| true` part says we want to ignore an error.   Why would we want to ignore an error?

Reply:
You may have applications that have gradlew, build.gradle, settings.gradle, gradle.properties ... the copy fails if one or more are not available. That piece of code is intended to copy all gradle files that the user might have in their project and not fail if one of them is not present.